__tests__/distributors/adoptium-installer.test.ts

```diff
@@ -0,0 +1,138 @@
+import { HttpClient } from '@actions/http-client';
+
+import * as semver from 'semver';
+
+import { AdoptiumDistributor } from '../../src/distributors/adoptium/installer';
+import { JavaInstallerOptions } from '../../src/distributors/base-models';
+
+let manifestData = require('../data/adoptium.json') as [];
+
+describe('getAvailableVersions', () => {
+  let spyHttpClient: jest.SpyInstance;
+
+  describe('build correct url', () => {
+    beforeEach(() => {
+      spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+      spyHttpClient.mockReturnValue({
+        statusCode: 200,
+        headers: {},
+        result: []
+      });
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+    });
+
+    it.each([
+      [
+        { version: '11', arch: 'x64', packageType: 'jdk' },
+        'os=mac&architecture=x64&image_type=jdk&release_type=ga&page_size=20&page=0'
+      ],
+      [
+        { version: '11', arch: 'x86', packageType: 'jdk' },
+        'os=mac&architecture=x86&image_type=jdk&release_type=ga&page_size=20&page=0'
+      ],
+      [
+        { version: '11', arch: 'x64', packageType: 'jre' },
+        'os=mac&architecture=x64&image_type=jre&release_type=ga&page_size=20&page=0'
+      ],
+      [
+        { version: '11-ea', arch: 'x64', packageType: 'jdk' },
+        'os=mac&architecture=x64&image_type=jdk&release_type=ea&page_size=20&page=0'
+      ]
+    ])(
+      'build correct url for %s',
+      async (installerOptions: JavaInstallerOptions, expectedParameters) => {
+        const distributor = new AdoptiumDistributor(installerOptions);
+        const baseUrl = 'https://api.adoptopenjdk.net/v3/assets/version/%5B1.0,100.0%5D';
+        const expectedUrl = `${baseUrl}?project=jdk&vendor=adoptopenjdk&heap_size=normal&jvm_impl=hotspot&sort_method=DEFAULT&sort_order=DESC&${expectedParameters}`;
+        distributor['getPlatformOption'] = () => 'mac';
+        await distributor['getAvailableVersions']();
+        expect(spyHttpClient.mock.calls).toHaveLength(1);
+        expect(spyHttpClient.mock.calls[0][0]).toBe(expectedUrl);
+      }
+    );
+  });
+
+  afterEach(() => {
```

Comment:
Is it intended to have this afterEach in the root of describe? it will run after the whole `describe('findPackageForDownload')` and not after each test in that suite

Reply:
We have internal afterEach and outer.

Reply:
Done. Thank you